File: QMP/QMPVOL.pas
unit
  QMPVOL;

interface

uses
  QMPNRM,
  QMPDCL;

type
  PQMPVOL = ^TQMPVOL;
  TQMPVOL = record
  private
    class var fnrm: TQMPNRM;
    class function Init(const Flags: Integer): Integer; cdecl; static;
    class procedure Quit(const Flags: Integer); cdecl; static;
    class function Modify(const Data: PData; const Latency: PInteger; const Flags: Integer): Integer; cdecl; static;
    class function Update(const Info: PInfo; const Flags: Integer): Integer; cdecl; static;
  public
    class function Plugin(): PPlugin; cdecl; static;
  end;

implementation

class function TQMPVOL.Plugin(): PPlugin; cdecl;
begin
  Result := New(PPlugin);
  Result.Description := 'Quinnware Normalizer v3.51';
  Result.Version := $0000;
  Result.Init := TQMPVOL.Init;
  Result.Quit := TQMPVOL.Quit;
  Result.Modify := TQMPVOL.Modify;
  Result.Update := TQMPVOL.Update;
end;

class function TQMPVOL.Init(const Flags: Integer): Integer; cdecl;
begin
  TQMPVOL.fnrm.Init();
  Result := 1;
end;

class procedure TQMPVOL.Quit(const Flags: Integer); cdecl;
begin
  TQMPVOL.fnrm.Done();
end;

class function TQMPVOL.Modify(const Data: PData; const Latency: PInteger; const Flags: Integer): Integer; cdecl;
begin
  TQMPVOL.fnrm.Process(Data^);
  Result := 1;
end;

class function TQMPVOL.Update(const Info: PInfo; const Flags: Integer): Integer; cdecl;
begin
  TQMPVOL.fnrm.Update(Info^);
  Result := 1;
end;

begin
end.

File: CPP/qmpvol.cpp
#pragma once
#include "qmpdcl.h"
#include "qmpvol.h"
#include "windows.h"

PPLUGIN QMPVOL::plugin() {
	PPLUGIN	result = new PLUGIN();

	result->description = L"Quinnware Normalizer v3.51";
	result->version = 0x0000;
	result->init = QMPVOL::init;
	result->update = QMPVOL::update;
	result->modify = QMPVOL::modify;

	return result;
};

INT QMPVOL::init(INT flags) {
	QMPVOL::nrm.init();

	return 1;
};

INT QMPVOL::modify(PDATA data, PINT latency, INT flags) {
	QMPVOL::nrm.process(*data);

	return 1;
};

INT QMPVOL::update(PINFO info, INT flags) {
	QMPVOL::nrm.update(*info);

	return 1;
};